cli/index_create.go

```diff
@@ -39,9 +41,33 @@ Example: create a named index for 'Users' collection on 'name' field:
 			store := mustGetContextStore(cmd)
 
 			var fields []client.IndexedFieldDescription
-			for _, name := range fieldsArg {
-				fields = append(fields, client.IndexedFieldDescription{Name: name})
+
+			for _, field := range fieldsArg {
+				var fieldName string
+				var order string
+				// For each field, parse it into a field name and ascension order, separated by a colon
+				// If there is no colon, assume the ascension order is ASC by default
+				const asc = "ASC"
+				const desc = "DESC"
+				parts := strings.Split(field, ":")
+				if len(parts) == 1 {
```

Comment:
suggestion: It looks like the way you have modified `wrapper_collection.go` makes it impossible to test this with our integration tests.  Please have a quick think and see if you can cover this without too much hassle.

Same goes for the `len(parts) > 2` block, but I see that as slightly less important.

Reply:
I proposed one [here](https://github.com/sourcenetwork/defradb/pull/3237/files#r1841233334), but I'm not sure if it's good enough

Reply:
thought: I defined a new error type, NewErrInvalidAscensionOrder which is returned if len(parts) > 2... because if that check is true, the fields parameter must be malformed. I thought this was a reasonable approach. Do we not think so?

Relevant comment also: [https://github.com/sourcenetwork/defradb/pull/3237#issuecomment-2474973523](url)

Reply:
100% agree with the prod code approach, my comment was only regarding the testing of it :)  It'll hit the same wrapper-problem in the discussion linked by Islam though, so not really worth worrying about in this PR IMO.

Reply:
suggestion(non-blocking): To remove one of the code paths, you could do something like this:
```go
const asc = "ASC"
const desc = "DESC"
parts := strings.Split(field, ":")
fieldName := parts[0]
order := asc
if len(parts) == 2 {
	order = strings.ToUpper(parts[1])
	if order != asc && order != desc {
		return NewErrInvalidAscensionOrder(field)
	}
} else if len(parts) > 2 {
	return NewErrInvalidAscensionOrder(field)
}
```

Reply:
you can't do `parts[0]` before checking the `len`

Reply:
> you can't do parts[0] before checking the len

Why not? `len(parts)` will never be zero.

Reply:
if you pass an empty string to `strings.Split` it will. This can happen for inputs "name,", ",name", "name,,age", ",", ",," ...

Reply:
The only way `strings.Split` will return a zero length slice is if both the string and the delimiter are empty `strings.Split("", "")`. This will never be true in this case. The delimiter is always `:` which means the length of the slice will always be at least 1.

https://go.dev/play/p/BQlVFauqCqn

Reply:
That makes sense to me. And I did run it to confirm, that yes even if you call it with a blank field, it won't cause a problem. Because instead you will get an error message returned:

Error: flag needs an argument: --fields

Although I think I am on board with Islam's idea of a `NewErrInvalidInxedFieldDescription` error for the case of one with a length > 2.

Reply:
I have added the new error type.